interfaces/builtin/mount_observe.go

```diff
@@ -45,6 +45,12 @@ const mountObserveConnectedPlugAppArmor = `
 owner @{PROC}/@{pid}/mounts r,
 owner @{PROC}/@{pid}/mountinfo r,
 owner @{PROC}/@{pid}/mountstats r,
+
+# some processes might read mount* from /proc/thread-self/ instead
+# and those resolve to the following: (no mountstats here)
+owner @{PROC}/@{pid}/task/@{tid}/mounts r,
+owner @{PROC}/@{pid}/task/@{tid}/mountinfo r,
```

Comment:
on a side note, I looked at the the base apparmor template, we have a bunch of entries that are also accessible under /proc/thread-self but are not accounted for:

```
     95:  owner @{PROC}/@{pid}/maps k,
    221:  owner @{PROC}/@{pid}/auxv r,
    233:  owner @{PROC}/@{pid}/cmdline r,
    234:  owner @{PROC}/@{pid}/comm r,
    260:  owner @{PROC}/@{pid}/cgroup rk,
    263:  owner @{PROC}/@{pid}/limits r,
    264:  owner @{PROC}/@{pid}/loginuid r,
    308:  owner @{PROC}/@{pid}/oom_{,score_}adj r,
```

cc @alexmurray

Reply:
@bboozzoo yep I think we should add the thread-self versions of these to the base template as well.